### DISCUSSION TITLE
How to collect the runtime application parameters?

### DISCUSSION THREAD

**Discussion by ender0229** [omnetpp/omnetpp#1071] [Q&A] [answered]
Thank you to the OMNeT++ working group：

 The topology might change during the simulation run, and similarly, applications might change too. Here, I am only concerned with applications used for sending traffic (such as UDP and TCP), which are affected by topology changes. For instance, if some nodes go offline, the traffic originally sent to those nodes can no longer be sent, leading to the termination of that application.
 How can I obtain the parameters of the applications sending traffic (such as source IP, destination IP, source MAC, destination MAC, sending interval, and transmission path) during the simulation run? Does OMNeT++ or INET provide any relevant examples, C++ classes, or APIs for this purpose?

**Comment by rhornig** [comment, marked as answer]
> Here, I am only concerned with applications used for sending traffic (such as UDP and TCP), which are affected by topology changes. 

Applications are not affected by topology changes.

> For instance, if some nodes go offline, the traffic originally sent to those nodes can no longer be sent, leading to the termination of that application.

No this is not true. If the destination node goes offline, the sending application will merrily continue without even noticing (e.g. an UDP app will just contune to send out the traffic). Tcp apps will also continue for a while until their send buffer if depleted, but they will not terminate. They will just start waiting for acknowledgements (until the end of simulation).

As for getting the parameter values, you can use `cModule::par()` to get any parameter value (as you usually use it in a simple module). As long as you have a pointer to a module, you can access its parameters however I don't see the reason why you would do that? Parameters would return the values that you set them in your INI file (or their default value). But you already know those values.

**Reply by ender0229** [reply]
Thank you very much.
